src/plugins/intel_gpu/src/graph/input_layout.cpp

```diff
@@ -59,7 +59,13 @@ void input_layout_inst::set_data(memory::ptr mem) {
 
     _has_valid_input = true;
     _output_changed = true;
-    _shape_changed = mem->get_layout() != ol;
+    auto check_shape_changed = [&]() {
+        if (ol.is_dynamic())
+            return mem->get_layout() != ol;
+        else
+            return mem->get_layout().get_layout_with_dims() != ol.get_layout_with_dims();
```

Comment:
I think this is quite confusing from an API point of view. 
How about implenting this handling in operator "==" of layout ?
Where we can check if it is dynamic, compare the layout as is, otherwise compare the shape aligned with the format.

Reply:
I reverted this commit (https://github.com/andrew-k-park/openvino/commit/664175c19211e1ab4e0e297e8784a33d6cc32750) and move partial shape comparison logic to equality operator in layout.